.jenkins/continuous.groovy

```diff
@@ -462,25 +462,26 @@ pipeline {
                         }
                     }
                 }
-                stage('CUDA-11.0.3-Clang-Tidy') {
+                stage('CUDA-11.8-Clang-15') {
                     agent {
                         dockerfile {
-                            filename 'Dockerfile.kokkosllvmproject'
+                            filename 'Dockerfile.nvcc'
                             dir 'scripts/docker'
                             label 'nvidia-docker && volta'
                             args '-v /tmp/ccache.kokkos:/tmp/ccache --env NVIDIA_VISIBLE_DEVICES=$NVIDIA_VISIBLE_DEVICES'
+                            additionalBuildArgs '--build-arg BASE=nvcr.io/nvidia/cuda:11.8.0-devel-ubuntu22.04 --build-arg ADDITIONAL_PACKAGES="clang-15 clang-tidy-15"'
                         }
                     }
                     steps {
                         sh 'ccache --zero-stats'
                         sh '''rm -rf build && mkdir -p build && cd build && \
                               cmake \
-                                -DCMAKE_BUILD_TYPE=Release \
-                                -DCMAKE_CXX_CLANG_TIDY="clang-tidy;-warnings-as-errors=*;--checks=-bugprone-unused-raii" \
+                                -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+                                -DCMAKE_CXX_CLANG_TIDY="clang-tidy-15;-warnings-as-errors=*" \
                                 -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
-                                -DCMAKE_CXX_COMPILER=clang++ \
-                                -DCMAKE_CXX_FLAGS="-Werror -Wno-unknown-cuda-version" \
-                                -DCMAKE_CXX_STANDARD=17 \
+                                -DCMAKE_CXX_COMPILER=clang++-15 \
+                                -DCMAKE_CXX_FLAGS="-Werror -Wno-unknown-cuda-version -Wno-pass-failed" \
```

Comment:
We are already suppressing a couple of warning in `Kokkos_SortImpl.hpp` and `Kokkos_SortByKeyImpl.hpp`. Any reason not to use
```
#pragma GCC diagnostic ignored "-Wpass-failed"
```
there?

Reply:
I think we should not test for this warning in our CI. It is purely a warning about some optimization not being able to be applied - and that is not something our pass/fail of CI should depend on.

Reply:
Merging as is.  We can revisit later.  I am sick of playing ping pong with the CI and this unblocks another PR of mine.